File: Harvest/Models/DailyQuiz.swift
import Foundation

struct DailyQuiz: Identifiable, Sendable {
    let id: String
    let trackingId: String
    let questionId: String
    var question: String
    var options: [QuizOption]
    var category: QuizCategory
    var selectedAnswer: String?
    var insight: String?
    let shownAt: String?
    var isAnswered: Bool
}

struct QuizOption: Codable, Hashable, Sendable, Identifiable {
    let id: String
    let text: String
}

enum QuizCategory: String, Codable, Sendable, CaseIterable {
    case datingStyle = "dating_style"
    case values
    case communication
    case relationshipGoals = "relationship_goals"
    case personality
}

File: android/app/src/main/java/com/harvestglass/harvest/data/model/DailyQuiz.kt
package com.harvestglass.harvest.data.model

import kotlinx.serialization.Serializable

/** Mirrors QuizCategory in Harvest/Models/DailyQuiz.swift. */
@Serializable
enum class QuizCategory(val raw: String) {
    DATING_STYLE("dating_style"),
    VALUES("values"),
    COMMUNICATION("communication"),
    RELATIONSHIP_GOALS("relationship_goals"),
    PERSONALITY("personality");

    companion object {
        /** Unknown categories fall back to dating style, as Swift's `?? .datingStyle` does. */
        fun fromRaw(raw: String?): QuizCategory =
            entries.firstOrNull { it.raw == raw } ?: DATING_STYLE
    }
}

@Serializable
data class QuizOption(val id: String, val text: String)

/** Mirrors DailyQuiz in Harvest/Models/DailyQuiz.swift. */
data class DailyQuiz(
    val id: String,
    val trackingId: String,
    val questionId: String,
    val question: String,
    val options: List<QuizOption>,
    val category: QuizCategory,
    val selectedAnswer: String? = null,
    val insight: String? = null,
    val shownAt: String? = null,
    val isAnswered: Boolean = false
)
